apps/rule-manager/client/src/ts/components/hooks/useRule.ts

```diff
@@ -112,6 +114,30 @@ export function useRule(ruleId: number | undefined) {
     return result;
   }
 
+  const unarchiveRule = async (ruleId: number) => {
+    setIsLoading(true);
+
+    const result = await fetch(`${location}rules/${ruleId}/unarchive`, {
+      method: 'POST',
+    }).then(responseHandler)
+
+    setIsLoading(false);
+
+    return result;
```

Comment:
At the moment, we're not updating the client representation of this rule (`[rule, setRule]`). We should do that – at the moment, we're closing the form, but (per another comment) I'm not convinced that we should, and if we don't close the form it'll be apparent that this state isn't being updated.

Per your suggestion, perhaps best to return `AllRuleData` and update `rule` in this way. It means we're always in sync with what the server thinks after our fetch completes.